site2/docs/client-libraries-cpp.md

```diff
@@ -4,298 +4,192 @@ title: Pulsar C++ client
-| [client-debuginfo](@pulsar:dist_rpm:client-debuginfo@) | [asc](@pulsar:dist_rpm:client-debuginfo@.asc),  [sha512](@pulsar:dist_rpm:client-debuginfo@.sha512) |
-| [client-devel](@pulsar:dist_rpm:client-devel@) | [asc](@pulsar:dist_rpm:client-devel@.asc),  [sha512](@pulsar:dist_rpm:client-devel@.sha512) |
-
-2. Install the package using the following command.
+2. Install the package using the following command:
 
 ```bash
 rpm -ivh apache-pulsar-client*.rpm
 ```
 
-After you install RPM successfully, Pulsar libraries are in the `/usr/lib` directory, for example：
-
-```bash
-lrwxrwxrwx 1 root root 18 Dec 30 22:21 libpulsar.so -> libpulsar.so.2.9.1
-lrwxrwxrwx 1 root root 23 Dec 30 22:21 libpulsarnossl.so -> libpulsarnossl.so.2.9.1
-```
+Now, you can see Pulsar C++ client libraries installed under the `/usr/lib` directory.
 
 :::note
 
-If you get the error that `libpulsar.so: cannot open shared object file: No such file or directory` when starting Pulsar client, you may need to run `ldconfig` first.
+If you get the error like "libpulsar.so: cannot open shared object file: No such file or directory" when starting Pulsar client, you need to run `ldconfig` first.
 
 :::
 
-2. Install the GCC and g++ using the following command, otherwise errors would occur in installing Node.js.
-
-```bash
-sudo yum -y install gcc automake autoconf libtool make
-sudo yum -y install gcc-c++
-```
-
-### Install Debian
-
-1. Download a Debian package from the links in the table.
-
-| Link | Crypto files |
-|------|--------------|
-| [client](@pulsar:deb:client@) | [asc](@pulsar:dist_deb:client@.asc), [sha512](@pulsar:dist_deb:client@.sha512) |
-| [client-devel](@pulsar:deb:client-devel@) | [asc](@pulsar:dist_deb:client-devel@.asc),  [sha512](@pulsar:dist_deb:client-devel@.sha512) |
+### Source
 
-2. Install the package using the following command.
+1. Clone the Pulsar repository and switch to the working directory `pulsar-client-cpp`:
 
 ```bash
-apt install ./apache-pulsar-client*.deb
+git clone https://github.com/apache/pulsar
+cd pulsar
+cd pulsar-client-cpp
 ```
 
-After you install DEB successfully, Pulsar libraries are in the `/usr/lib` directory.
-
-### Build
-
-If you want to build RPM and Debian packages from the latest master, follow the instructions below. You must run all the instructions at the root directory of your cloned Pulsar repository.
-
-There are recipes that build RPM and Debian packages containing a
-statically linked `libpulsar.so` / `libpulsarnossl.so` / `libpulsar.a` / `libpulsarwithdeps.a` with all required dependencies.
+2. Install dependencies:
 
-To build the C++ library packages, you need to build the Java packages first.
+<Tabs>
+<TabItem value="Linux">
 
-```shell
-mvn install -DskipTests
-```
+```bash
+apt install cmake libssl-dev libcurl4-openssl-dev liblog4cxx-dev libprotobuf-dev protobuf-compiler libboost-all-dev google-mock libgtest-dev libjsoncpp-dev
 
-#### RPM
+# libgtest-dev version is 1.18.0 or above
```

Comment:
Do users of this c++ client need gtest? Or only developers of the c++ client itself need it?

Reply:
Good point. I suppose only the client developers should build with test. Don't know whether we have a hard dependency in release mode. cc @BewareMyPower

Reply:
We can use `-DBUILD_TESTS=OFF` to disable that

Reply:
@merlimat Thanks for your information! I made this commit https://github.com/apache/pulsar/pull/17580/commits/2e9a946840194b6f46e39e1b75c0887c956cfe2d for excluding tests. Please take a look.